### PR TITLE
Support SNI in duo client

### PR DESCRIPTION
**Pull request by ccassell**
Connecting to a service that uses SSL/TLS with SNI, for example AWS CloudFront, raises an `SSLV3_ALERT_HANDSHAKE_FAILURE` exception. To fix this, pass server hostname to `SSLContext.wrap_socket`.

### CODE REVIEW

File: duo_client/https_wrapper.py [hunk 1]
```diff
@@ -118,7 +118,8 @@ def connect(self):
                                              self.timeout)
         if self._tunnel_host:
             self._tunnel()
-        self.sock = self.default_ssl_context.wrap_socket(self.sock)
+        self.sock = self.default_ssl_context.wrap_socket(self.sock,
+                                                         server_hostname=self.host)
```
Comment (AaronAtDuo) [inline duo_client/https_wrapper.py:122]: Do we need to do any check on self.host?  that it's not None for instance?

Comment (AaronAtDuo) [reply duo_client/https_wrapper.py:122]: Line 63 says it can be "host:port" - would we need to strip off the port piece here?

Comment (ccassell) [reply duo_client/https_wrapper.py:122]: I just added a test case for for host:port form. Checking for None isn't necessary: the constructor excepts if host is None

**Review by AaronAtDuo — APPROVED** [review]

**Review by yizshi — APPROVED** [review]

**Comment by AaronAtDuo** [comment]
thanks @ccassell

**Pull request merged**